pkg/kubelet/kubelet.go

```diff
@@ -3671,6 +3671,14 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *api.Pod, podStatus *kubeco
 
 	// Handle the containers failed to be started, which should be in Waiting state.
 	for _, container := range containers {
+		if hasInitContainers {
```

Comment:
The code above uses `hasInitContainers`, so I'm not sure which to use...

/cc @smarterclayton